src/target/riscv/riscv-013.c

```diff
@@ -686,6 +693,30 @@ static int dmi_write_exec(struct target *target, uint32_t address,
 	return dmi_op(target, NULL, NULL, DMI_OP_WRITE, address, value, true, ensure_success);
 }
 
+static bool check_dbgbase_exists(struct target *target)
+{
+	uint32_t dbgbase_to_find = target->dbgbase;
+	/* Start with the first debug module */
+	uint32_t nextdm = 0;
+	uint32_t count = 0;
+
+	LOG_TARGET_DEBUG(target, "Searching for DM with DMI base address (dbgbase) = 0x%x", target->dbgbase);
+	while (1) {
+		target->dbgbase = nextdm;
+		if (target->dbgbase == dbgbase_to_find)
+			return true;
+		if (dmi_read(target, &nextdm, DM_NEXTDM) != ERROR_OK)
+			break;
+		LOG_TARGET_DEBUG(target, "dm @ 0x%x --> nextdm=0x%x", target->dbgbase, nextdm);
+		if (nextdm == 0) /* last one in the chain */
+			break;
+		if (count++ >= RISCV_MAX_DMS)
+			break;
```

Comment:
Here you may add a comment, explaining why there is the loop counter - it may not be clear to the reader.

```suggestion
		/* Safety: Avoid looping forever in case of buggy nexdm values in the hardware. */
		if (count++ >= RISCV_MAX_DMS)
			break;
```

Reply:
done